.github/workflows/tests-e2e-upgrade.yaml

```diff
@@ -654,15 +654,39 @@ jobs:
     runs-on: ubuntu-latest
     needs: setup-and-test
     steps:
+      # TODO remove list and filter artifact steps if this PR https://github.com/actions/upload-artifact/pull/521 is merged
+      - name: List All Artifacts
+        id: list_artifacts
+        run: |
+          echo "Fetching artifacts"
+          curl --fail --show-error --silent \
+               -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+               -H "Accept: application/vnd.github.v3+json" \
+               "https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/artifacts" \
+               | jq -r '.artifacts[] | .name' > artifacts_list.json
+      - name: Filter Artifacts
+        id: filter_artifacts
+        run: |
+          echo "Filtered Artifacts starts with cilium-sysdumps:"
+          if grep -E "^cilium-sysdumps-" artifacts_list.json; then
+            echo "sysdumps=true" >> $GITHUB_OUTPUT
+          else
+            echo "::warning::No matching cilium-sysdumps-* artifacts found."
+          fi
+          echo "Filtered Artifacts starts with cilium-junits:"
+          if grep -E "^cilium-junits-" artifacts_list.json; then
+            echo "junits=true" >> $GITHUB_OUTPUT
```

Comment:
This variable doesn't seem to be used anywhere. Is it supposed to be checked in the `Merge JUnits` step?

Reply:
I will make a follow-up PR, thanks.

Reply:
Nice catch, I am in your dept 🙏 This proves your point on reusable workflows.